node/lib/util/hook.js

```diff
@@ -67,6 +67,8 @@ exports.execHook = co.wrap(function*(repo, name, args=[]) {
         } else if (e.stdout) {
             console.log(e.stdout);
             return e.stdout;
+        } else if (e.code === 1) {
```

Comment:
The code might be 2, or 143, and we should still abort the commit, right?  Also, this function previously returned a string and now it sometimes returns a string and sometimes and int, which seems really confusing.

Reply:
I agree it is confusing. I made it return a number only because the JSDoc specified so. I'd be happy to change this to however you think it makes sense. Please let me know exactly what you want me to do. Thanks.

Reply:
Figure out why it's returning a string -- does something depend on that?  If so, I guess we have to keep it a string, and fix the javadoc.  Then replace  ```if (e.code === 1)  { return 1;}``` with ```return "hook failed with no output"``` or something.  

I would be somewhat surprised if the current code worked at all, because stdio is set to 'inherit', so we should not we be able to get stdout?  And anyway we probably couldn't get it from the error object... (I can't tell for sure because the child-process-promise docs are anemic and the child-process docs area lso pretty lacking.

Reply:
I did a quick search in the code base and did not see the return of Hook.execHook being used. Looking at the code of `child-process-promise`:
https://github.com/patrick-steele-idem/child-process-promise/blob/master/lib/index.js
The result.status is never set and so Hook.execHook would return undefined when spawn succeeded. I suspect it meant to return `result.code`. And I agree with you that without using the [capture](https://github.com/patrick-steele-idem/child-process-promise#capture) option, the current code should not work at all.

Reply:
I think fixing the bigger problem of this function is out of the scope of this PR. I pushed a change to return `"hook failed with no output"` as you suggested.

Reply:
You are still checking that the code === 1.  Don't do that -- just unconditionally return that message.

Reply:
done